apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1728,13 +1727,31 @@ default_report_flags() ->
 
 report(Tag, St, D) -> report_info(do_rpt(Tag, D), Tag, St, D).
 
-report_info(DoRpt, Tag, St, #data{client = Client, channel_id = ChanId}) ->
-    lager:debug("report_info(~p, ~p, ~p, ~p)", [DoRpt, Tag, ChanId, St]),
-    if DoRpt -> Client ! {?MODULE, self(), ChanId, {Tag, St}};
-       true  -> ok
+report_info(DoRpt, Tag, Info, #data{client = Client} = D) ->
+    if DoRpt ->
+            Msg = rpt_message(report, Tag, Info, D),
+            lager:debug("report_info(true, ~p)", [Msg]),
+            Client ! {?MODULE, self(), Msg};
+       true  ->
+            lager:debug("report_info(~p, ~p, ~p)", [DoRpt, Tag, Info]),
+            ok
     end,
     ok.
 
+rpt_message(Type, Tag, Info, #data{on_chain_id = undefined,
+                                   channel_id  = TmpChId}) ->
+    #{ temp_channel_id => TmpChId
```

Comment:
I see this is not used by client. Shall we addapt the WS to propagate it some how?

Reply:
Perhaps best to simply remove it if it won't break anything.